google-cloud-storage/src/test/java/com/google/cloud/storage/it/ITReadMaskTest.java

```diff
@@ -0,0 +1,446 @@
+import com.google.cloud.storage.conformance.retry.ParallelParameterized;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.experimental.runners.Enclosed;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized.Parameters;
+
+@RunWith(Enclosed.class)
+public final class ITReadMaskTest {
+
+  @ClassRule(order = 1)
+  public static final StorageFixture sfH = StorageFixture.defaultHttp();
+
+  @ClassRule(order = 1)
+  public static final StorageFixture sfG = StorageFixture.defaultGrpc();
+
+  @ClassRule(order = 2)
+  public static final BucketFixture bucketFixture =
+      BucketFixture.newBuilder()
+          .setHandle(sfH::getInstance)
+          .setBucketNameFmtString("java-storage-grpc-%s")
+          .build();
+
+  private static Storage sh;
+  private static Storage sg;
+  private static BlobId blobId;
+
+  @BeforeClass
+  public static void beforeClass() {
+    sh = sfH.getInstance();
+    sg = sfG.getInstance();
+
+    String bucket = bucketFixture.getBucketInfo().getName();
+
+    BlobId blobId1 = BlobId.of(bucket, objName("001"));
+    BlobId blobId2 = BlobId.of(bucket, objName("002"));
+    BlobId blobId3 = BlobId.of(bucket, objName("003"));
+    BlobId blobId4 = BlobId.of(bucket, objName("004"));
+
+    BlobInfo info1 = BlobInfo.newBuilder(blobId1).setMetadata(ImmutableMap.of("pow", "1")).build();
+    BlobInfo info2 = BlobInfo.newBuilder(blobId2).setMetadata(ImmutableMap.of("pow", "2")).build();
+    BlobInfo info3 = BlobInfo.newBuilder(blobId3).setMetadata(ImmutableMap.of("pow", "3")).build();
+    BlobInfo info4 = BlobInfo.newBuilder(blobId4).setMetadata(ImmutableMap.of("pow", "4")).build();
+    sh.create(info1, "A".getBytes(StandardCharsets.UTF_8), BlobTargetOption.doesNotExist());
+
+    ComposeRequest c2 =
+        ComposeRequest.newBuilder()
+            .addSource(blobId1.getName(), blobId1.getName())
+            .setTarget(info2)
+            .setTargetOptions(BlobTargetOption.doesNotExist())
+            .build();
+    ComposeRequest c3 =
+        ComposeRequest.newBuilder()
+            .addSource(blobId2.getName(), blobId2.getName())
+            .setTarget(info3)
+            .setTargetOptions(BlobTargetOption.doesNotExist())
+            .build();
+    ComposeRequest c4 =
+        ComposeRequest.newBuilder()
+            .addSource(blobId3.getName(), blobId3.getName())
+            .setTarget(info4)
+            .setTargetOptions(BlobTargetOption.doesNotExist())
+            .build();
+    sh.compose(c2);
+    sh.compose(c3);
+    sh.compose(c4);
+
+    blobId = blobId1;
+  }
+
+  /*
+  public static final class UpdateMask {
```

Comment:
reminding you of this that you said you would remove during our pair session.

Reply:
Thank you, removed.